applicationset/generators/scm_provider_test.go

```diff
@@ -200,3 +200,89 @@ func TestSCMProviderGenerateParams(t *testing.T) {
 		})
 	}
 }
+
+func TestAllowedSCMProvider(t *testing.T) {
+	cases := []struct {
+		name           string
+		providerConfig *argoprojiov1alpha1.SCMProviderGenerator
+		expectedError  string
+	}{
+		{
+			name: "Error Github",
+			providerConfig: &argoprojiov1alpha1.SCMProviderGenerator{
+				Github: &argoprojiov1alpha1.SCMProviderGeneratorGithub{
+					API: "myservice.mynamespace.svc.cluster.local",
```

Comment:
Do we need to concern ourselves with schemas? i.e. can a user specify `https://`, `http://`, etc?

Reply:
When I did not put the http or HTTPS during my tests, an error was raised with "unknown schema"
I should have put the http here in the test